yt_dlp/extractor/boosty.py

```diff
@@ -160,18 +163,39 @@ def _extract_formats(self, player_urls, video_id):
                 self.report_warning(f'Unknown format type: {format_type!r}')
         return formats
 
+    def _extract_posts_from_html(self, url, post_id, webpage=None):
```

Comment:
There is no reason to have two ways to get the post data. Since we have to extract the data from the page for paywalled posts anyway, we can _always_ extract the data from the page and remove API-related code altogether.

Reply:
true, my thought was that scraping the website may be more likely to break in the future so it would be better to only have it break for paywalled posts

it seems like people in #9401 have found a way to use the api, i'll try that out